packages/retail-ui-extensions/src/extension-api/types/multiple-resource-result.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * The result of a fetch function where the input is multiple IDs.
+ * This object contains the resources that were found, as well as an array of IDs specifying which IDs, if any, did not correspond to a resource.
+ */
+export interface MultipleResourceResult<T> {
+  /**
+   * The resources that were fetched using the IDs provided.
+   */
+  fetchedResources?: T[];
+  /**
+   * The IDs for which a resource was not found.
+   */
+  idsNotFound?: number[];
```

Comment:
I'd rather keep it simple for now, as now we're adding scope by defining what the reasons can be. I think specifying that POS couldn't find a resource for the given ID is good enough. Plus I feel like POS doesn't really know either, it might be a mistake with sync, it could be that the product is not published on POS, but as far as POS understands, it's just not in its DB